src/Icicle/Storage/Dictionary/Toml.hs

```diff
@@ -135,11 +138,11 @@ loadImports
   -> EitherT DictionaryImportError IO FunEnvT
 loadImports parentFuncs parsedImports
  = hoistEither . mapLeft DictionaryErrorCheck
- $ foldlM go parentFuncs parsedImports
+ $ foldlM (go parentFuncs) [] parsedImports
  where
-  go env f
-   = do f' <- P.sourceDesugarF f >>= P.sourceCheckF
-        return $ M.union f' env
+  go env acc f
+   = do f' <- P.sourceCheckF (env <> acc) f
+        return $ acc <> f'
```

Comment:
Ok.
And Tran, yeah, we don't need to carry the desugared version along, and this way when we persist the dictionary it looks pretty much identical.